src/test/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSourceDeserializationTest.java

```diff
@@ -0,0 +1,23 @@
+package io.jenkins.plugins.gitlabbranchsource;
+
+import com.thoughtworks.xstream.XStream;
+import java.lang.reflect.Field;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+
+public class GitLabSCMSourceDeserializationTest {
+
+    @Test
+    public void afterDeserializationWithXStreamTransientFieldsAreNotNull() throws Exception {
+        XStream xs = new XStream();
+        GitLabSCMSource source = (GitLabSCMSource) xs.fromXML(xs.toXML(new GitLabSCMSource("Test Server", "Test Owner", "test-path")));
```

Comment:
Not that the test is not good but could be helpful to validate with a Restartable Jenkins Rule instead.

Reply:
You are right. Re-implemented test to facilitate `RestartableJenkinsRule`.